### ISSUE TITLE
Custom iteration callback?

### ISSUE THREAD

**Issue by tristanz** [catboost/catboost#147]
Is there a way to specify a custom iteration callback which gets the current loss?  I'd like to track this externally in realtime if possible.

**Comment by annaveronika** [comment]
Loss function values are written to files during training, you could read them. https://tech.yandex.com/catboost/doc/dg/concepts/output-data_loss-function-docpage/

**Comment by vroomzel** [comment]
Here's a good example where a callback feature becomes extremely important: https://github.com/ray-project/ray/blob/master/python/ray/tune/examples/lightgbm_example.py#L31
I'd like to be able to use CatBoost with Ray's Tune framework in place of XgBoost and LightGBM, but reading loss function values from file won't work (at least it's not straightforward to implement)

**Comment by neomatrix369** [comment]
I have used the `allow_writing_files=True` flag and it the below happens:

Folder by the name `catboost_info` is created which contains the below:

```
catboost_info
├── catboost_training.json
├── learn
│   └── events.out.tfevents
├── learn_error.tsv
├── test
│   └── events.out.tfevents
├── test_error.tsv
└── time_left.tsv
```

Any additional flags to add/enabled to get additional metrics when an experiment is running?

Thanks for this https://catboost.ai/docs/concepts/output-data_loss-function.html, I think similar params can be found on https://catboost.ai/docs/concepts/python-reference_parameters-list.html although usage isn't very clear without an example or two.
